src/plans/conjugate_gradient_plan.jl

```diff
@@ -112,21 +116,23 @@ function ConjugateGradientDescentState(
     M::AbstractManifold;
     p::P=rand(M),
     coefficient::Union{DirectionUpdateRule,ManifoldDefaultsFactory}=ConjugateDescentCoefficient(),
-    retraction_method::AbstractRetractionMethod=default_retraction_method(M, typeof(p)),
-    stepsize::Stepsize=default_stepsize(
+    restart_condition::TRC=NeverRestart(),
+    retraction_method::TRetr=default_retraction_method(M, typeof(p)),
+    stepsize::TStep=default_stepsize(
         M, ConjugateGradientDescentState; retraction_method=retraction_method
     ),
-    stopping_criterion::StoppingCriterion=StopAfterIteration(500) |
+    stopping_criterion::TsC=StopAfterIteration(500) |
                                           StopWhenGradientNormLess(1e-8),
-    vector_transport_method=default_vector_transport_method(M, typeof(p)),
+    vector_transport_method::VTM=default_vector_transport_method(M, typeof(p)),
     initial_gradient::T=zero_vector(M, p),
-) where {P,T}
-    return ConjugateGradientDescentState{P,T}(
+) where {P,T,TsC<:StoppingCriterion,TStep<:Stepsize,TRC<:AbstractRestartCondition,TRetr<:AbstractRetractionMethod,VTM<:AbstractVectorTransportMethod}
+    return ConjugateGradientDescentState{P,T,TsC,TStep,TRC,TRetr,VTM}(
```

Comment:
```suggestion
    return ConjugateGradientDescentState(
```
missed to remove it here (and have to add those bz comment anyways).